internal/pkg/cli/clean/static_site.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package clean
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/copilot-cli/internal/pkg/s3"
+)
+
+type bucketResourceGetter interface {
+	BucketName(app, env, wkld string) (string, error)
+}
+
+type bucketEmptier interface {
+	EmptyBucket(string) error
+}
+
+// StaticSite is used to clean up resources created for a static site.
+type StaticSite struct {
+	app, env, svc        string
+	bucketResourceGetter bucketResourceGetter
+	bucketEmptier        bucketEmptier
+}
+
+// NewStaticSite returns an initialized static site.
+func NewStaticSite(app, env, svc string, rg bucketResourceGetter, emptier bucketEmptier) *StaticSite {
```

Comment:
nit: I wonder if the names should be like this:
```suggestion
type StaticSiteCleaner struct {
	app, env, svc        string
	bucketResourceGetter bucketResourceGetter
	bucketEmptier        bucketEmptier
}

// StaticSite returns an initialized static site cleaner.
func StaticSite(app, env, svc string, rg bucketResourceGetter, emptier bucketEmptier) *StaticSite {
```

so that it reads: 

```go
cleaner := clean.StaticSite()
cleaner.Clean()
```